src/lib.rs

```diff
@@ -180,15 +180,55 @@ pub trait Specifier {
     ///
     /// This is the type that is used internally for computations.
     type Base: Default + private::PushBits + private::PopBits;
+
+    /// The type returned by getter functions.
+    ///
+    /// # Note
+    ///
+    /// Although this is usually the same as the Face type, in the case of
+    /// non-power-of-two enums (or other fields that may fail), this type is
+    /// a Result type.
+    type GetterReturn;
+
     /// The interface type of the specifier.
     ///
     /// # Note
     ///
     /// This is the type that is used for the getters and setters.
-    type Face: private::FromBits<Self::Base> + private::IntoBits<Self::Base>;
+    type Face: private::FromBits<Self::Base, Self::GetterReturn>
+        + private::IntoBits<Self::Base>;
 }
 
 /// The default set of predefined specifiers.
 pub mod specifiers {
     ::modular_bitfield_impl::define_specifiers!();
 }
+
+/// Error type used for working with enums
+#[derive(Debug, PartialEq)]
```

Comment:
Although I like the simplicity, I think we can be more generous here with providing some more out-of-the-box derives such as `Eq`, `Copy` and `Clone` and maybe also `PartialOrd` and `Ord`.

Reply:
`Clone` and `Eq`, I agree with. `PartialOrd` and `Ord` I'm not sure make sense (for the underlying raw value, yes, but sorting errors doesn't have a use case I can think of).

Ditto with `Copy`, I think the user should just extract the raw value if they want to do copying with it.